cookbooks/planet/templates/default/planetdump.erb

```diff
@@ -146,20 +167,41 @@ function install_dump {
   local year="$5"
   local name="${type}-${date}.osm.${format}"
   local latest="${type}-latest.osm.${format}"
-  local rss_file="${type}-${format}-rss.xml"
+  local rss_file="${type}-${format}-rss.xml"  
+
+  # Check if the files exist before proceeding
+  test -f "${name}" || { echo "Error: File ${name} not found!"; exit 1; }
+  test -f "${name}.torrent" || { echo "Error: File ${name}.torrent not found!"; exit 1; }
+  test -f "${rss_file}" || { echo "Error: RSS file ${rss_file} not found!"; exit 1; }
 
+  # Generate the MD5 checksum for the dump
   md5sum "${name}" > "${name}.md5"
+
+  # Upload all files to S3
+  /opt/awscli/v2/current/bin/aws --profile=osm-pds-upload s3 cp --storage-class=INTELLIGENT_TIERING --no-progress "${name}.md5" "${s3dir}/${name}.md5"
+  /opt/awscli/v2/current/bin/aws --profile=osm-pds-upload s3 cp --storage-class=INTELLIGENT_TIERING --no-progress "${name}.torrent" "${s3dir}/${name}.torrent"
+  /opt/awscli/v2/current/bin/aws --profile=osm-pds-upload s3 cp --storage-class=INTELLIGENT_TIERING --no-progress "${name}" "${s3dir}/${name}"
+
+  # Wait for the replication of each uploaded file on S3
+  replication_status_check "https://osm-planet-eu-central-1.s3.amazonaws.com/${s3dir}/${name}.md5"
+  replication_status_check "https://osm-planet-eu-central-1.s3.amazonaws.com/${s3dir}/${name}.torrent"
+  # the dump is expected to take longer than the md5 and torrent, and gets checked last.
+  replication_status_check "https://osm-planet-eu-central-1.s3.amazonaws.com/${s3dir}/${name}"
+
+  # Ensure the target directory exists
   mkdir -p "${dir}/${year}"
-  mv "${name}" "${name}.md5" "${dir}/${year}"
-  ln -sf "${year:-.}/${name}" "${dir}/${latest}"
-  test -f "${name}.torrent" && mv "${name}.torrent" "${dir}/${year}" && ln -sf "${year:-.}/${name}.torrent" "${dir}/${latest}.torrent"
-  test -f "${rss_file}" && xmllint --noout "${rss_file}" && cp -f "${rss_file}" "${dir}"
+  # Move the dump and its MD5 checksum to the target directory
+  mv "${name}" "${name}.md5" "${dir}/${year}" || { echo "Error moving files to ${dir}/${year}"; exit 1; }
+  # Create or update the symbolic link for the latest dump
+  ln -sf "${name}" "${dir}/${latest}" || { echo "Error creating symbolic link for ${name}"; exit 1; }
```

Comment:
why ${year:-.}/ gone?

Reply:
As far as I can tell, `${year:-.}` is always empty here?

Reply:
If it then something has changed - the two PBF calls to that function don't set it currently as they don't files by year in the local store.